cmd/snap-exec/main_test.go

```diff
@@ -242,3 +242,52 @@ foobar
 
 `)
 }
+
+func (s *snapExecSuite) TestSnapExecHookRealIntegration(c *C) {
+	// we need a lot of mocks
```

Comment:
Looks like a lot is repeated between these tests. What amount of logic is unique, and would it be easy to put that core test logic into a function that is passed into another function, instead of blindly copy & pasting all the boilerplate every time?